src/LocalNeuralNetImp.hpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef LOCALNEURALNETIMP_HPP_INCLUDE
+#define LOCALNEURALNETIMP_HPP_INCLUDE
+
+#include "LocalNeuralNet.hpp"
+
+namespace geopm
+{
+    class LocalNeuralNetImp : public LocalNeuralNet
+    {
+        public:
+            LocalNeuralNetImp(std::vector<std::pair<std::vector<std::vector<float> >, std::vector<float> > > input);
+            std::vector<float> model(std::vector<float> inp);
+
+        private:
+            /// @brief Class to store and perform operations on 1D Tensors,
+            ///        aka vectors, suitable for use in feed-forward neural
+            ///        networks.
+            class TensorOneD
```

Comment:
Q: is there a reason why you opted to make the tensor classes internal?

If they are really particular to LocalNeuralNetImp, then it makes sense. Otherwise, my concern is that this causes unnecessary back-and-forth conversion between the STL types and the tensor classes every time you want to interact with a LocalNeuralNetImp object.

Reply:
This is what @cmcantalupo and I (@alawibaba) came up with -- totally happy to have a discussion about this.